repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -414,11 +425,10 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         MDC.put(TracingKey.TRACING_KEY, traceGUID);
 
         try {
-            try {
-                // ensures that the method name exists
-                HttpComponentFactory.valueOf(wrappedRequest.getMethod().toUpperCase());
-            } catch (IllegalArgumentException iae) {
-                throw new InvalidMethodException("Request contained an unknown method.", iae);
+            // Ensures that the method name is supported
+            // todo: HTTP request methods are case-sensitive, so this check should not upper case the request method
+            if (!SUPPORTED_HTTP_METHODS.contains(wrappedRequest.getMethod().toUpperCase())) {
+                throw new InvalidMethodException(wrappedRequest.getMethod() + " method not supported");
```

Comment:
I ended up leaving this check in for backwards compatibility. This is no longer an Apache HttpClient constraint, but purely a Repose constraint.

I actually attempted to drop this check since we are no longer forced to support a subset of methods, but `CONNECT` gave me some issues. It's a method meant to work with proxies specifically, and Repose certainly does not support it. If desired, I could just blacklist `CONNECT` rather than whitelisting the old list of supported methods.

Reply:
leave the list as is for now.